src/test/java/com/krech/botv3/WordServiceTest.java

```diff
@@ -16,56 +21,45 @@
 import static org.mockito.Mockito.*;
 
 
-
-
 @ExtendWith(MockitoExtension.class) // ExtendWith - Juint; MockitoExtension - mockito. эта строка говорит junit'у, что нужно подключтить мокито к тестам, и тогда у нас заработают аннотации @Mock @InjectMocks
 class WordServiceTest {
 
     @Mock // Mockito поля, помеченные этой аннотацией инициализируются с помощью Mockito. Изначально это пустые объекты болванки, которые ничего не делают, все методы возвращают null или пустые коллекции(стоит проверить)
     WordRepository wordRepository;
 
-    @Mock
-    InMemoryIndexRepositoryImpl inMemoryIndexRepositoryImpl;
 
     @Mock
-    Indexkey indexkey;
+    IndexRepository indexRepository;
+
 
     WordService wordService;
 
+
     @BeforeEach //Junit - метод, в еоторый пожно поместитть некоторый пресетап для каждого теста в классе
-    public void init(){
-        wordService = new WordService(inMemoryIndexRepositoryImpl, wordRepository);
+    public void init() {
+        wordService = new WordService(indexRepository, wordRepository);
     }
 
     @Test
-    void searchWordsForClientTest(){
+    void searchWordsForClientTest() {
         String str = "Феиксн";
 
-        List <String> expected = new ArrayList<>();
+        List<String> expected = new ArrayList<>();
         expected.add("Фениксятина");
         expected.add("Феникс");
 
+        List <IndexObject> listOfIndexObject = new ArrayList<>();
+
 
-        HashMap<Indexkey, Set<String>> indexStorage = new HashMap<>();
-        Indexkey indexkey = new Indexkey('Ф', "еникс");
-        Indexkey indexkey1 = new Indexkey('Ф',"ениксятина");
-        Set <String> setWithoutNumber = new HashSet<>();
-        Set <String> set1 = new HashSet<>();
 
-        setWithoutNumber.add("Феникс");
-        setWithoutNumber.add("Фениксятина");
-        set1.add("Феникс");
-        set1.add("Фениксятина");
-        set1.add("Фениксятиноподобный");
-        set1.add("Фениксблаблабла");
-        set1.add("Фениксовость");
+        WordObject fenix = new WordObject("Феникс", "Ф");
+        WordObject fenixyatina = new WordObject("Фениксятина", "Ф");
 
-        indexStorage.put(indexkey,setWithoutNumber);
-        indexStorage.put(indexkey1,set1);
+        when(indexRepository.findByFirstLetter(eq(String.valueOf(str.charAt(0))))).thenReturn(listOfIndexObject);
+        when(wordRepository.findByFirstLetter(eq(String.valueOf(str.charAt(0))))).thenReturn(List.of(fenix, fenixyatina));
 
-        when(inMemoryIndexRepositoryImpl.getAll()).thenReturn(indexStorage);
 
-        List <String> words = wordService.searchWordsForClient(str);
+        List<String> words = wordService.searchWordsForClient(str);
 
         assertEquals(expected.size(), words.size());
```

Comment:
It would be nice to initialize the "expected" array right before assertion. It will make code more readable. Also since java 11 you can do initialization in modern style: List<String> expected = List.of("Фениксятина", "Феникс");